frontend/src/pages/notebookController/SpawnerPage.tsx

```diff
@@ -29,30 +29,34 @@ import {
   usernameTranslate,
 } from '../../utilities/utils';
 import AppContext from '../../app/AppContext';
-import { ODH_NOTEBOOK_REPO } from '../../utilities/const';
-import NotebookControllerContext from './NotebookControllerContext';
+import { FAST_POLL_INTERVAL, ODH_NOTEBOOK_REPO, POLL_INTERVAL } from '../../utilities/const';
 import { getGPU } from '../../services/gpuService';
 import { patchDashboardConfig } from '../../services/dashboardConfigService';
 import { createSecret, getSecret, replaceSecret } from '../../services/secretsService';
+import { useWatchNotebook } from '../../utilities/useWatchNotebook';
+import { useWatchImages } from '../../utilities/useWatchImages';
+import ApplicationsPage from '../../pages/ApplicationsPage';
+import StartServerModal from './StartServerModal';
 
 import './NotebookController.scss';
 
-type SpawnerPageProps = {
-  setStartModalShown: (shown: boolean) => void;
-  updateNotebook: () => void;
-  setNotebookPollInterval: (interval: number) => void;
-};
-
-const SpawnerPage: React.FC<SpawnerPageProps> = React.memo(({ setStartModalShown }) => {
+const SpawnerPage: React.FC = React.memo(() => {
   const history = useHistory();
-  const { images, dashboardConfig } = React.useContext(NotebookControllerContext);
-  const [username, namespace] = useSelector<State, [string, string]>((state) => [
-    state.appState.user || '',
-    state.appState.namespace || '',
-  ]);
+  const { images, loaded, loadError } = useWatchImages();
+  const { username } = useParams<{ username: string }>();
+  const namespace = useSelector<State, string>((state) => state.appState.namespace || '');
   const projectName = ODH_NOTEBOOK_REPO || namespace;
+  const {
+    notebook,
+    loaded: notebookLoaded,
+    setPollInterval: setNotebookPollInterval,
+  } = useWatchNotebook(projectName);
```

Comment:
Actually, can you take this hook and wrap it? I'd like you to start thinking of ways to keep like-minded chunks of code together and not pollute components with just "facts of life".

You create a `useEffect` below that sets the poll interval... this is more like a hook called `useWatchNotebookForSpawner` where you do both of those together as a hook, and return `notebook` and `notebookLoaded` to the Spawner component.

The Spawner component does not need to know of the special nature of reloading of the notebook... so we can move that out a layer and thus clean up the Spawner page a bit.

Thoughts?